Philips.CodeAnalysis.MsTestAnalyzers/TestMethodsMustHaveTheCorrectNumberOfArgumentsAnalyzer.cs

```diff
@@ -38,25 +39,35 @@ public TestMethodsMustHaveTheCorrectNumberOfArguments(MsTestAttributeDefinitions
 			protected override void OnTestMethod(SyntaxNodeAnalysisContext context, MethodDeclarationSyntax methodDeclaration, IMethodSymbol methodSymbol, bool isDataTestMethod)
 			{
 				int? expectedNumberOfParameters;
-				if (!isDataTestMethod)
-				{
-					expectedNumberOfParameters = 0;
-				}
-				else
+				if (isDataTestMethod)
 				{
+					if (HasParams(methodDeclaration))
+					{
+						return;
+					}
+
 					if (!TryGetExpectedParameters(methodDeclaration, context, out expectedNumberOfParameters))
 					{
 						context.ReportDiagnostic(Diagnostic.Create(Rule, methodDeclaration.Identifier.GetLocation(), methodDeclaration.Identifier));
 						return;
 					}
 				}
+				else
+				{
+					expectedNumberOfParameters = 0;
+				}
 
 				if (expectedNumberOfParameters != null && expectedNumberOfParameters != methodDeclaration.ParameterList.Parameters.Count)
 				{
 					context.ReportDiagnostic(Diagnostic.Create(Rule, methodDeclaration.Identifier.GetLocation(), methodDeclaration.Identifier));
 				}
 			}
 
+			private bool HasParams(MethodDeclarationSyntax methodDeclaration)
+			{
+				return methodDeclaration.ParameterList.Parameters.LastOrDefault()?.Modifiers.Any(SyntaxKind.ParamsKeyword) == true;
```

Comment:
== true is not required

Reply:
Required to avoid propagating nullable bool